src/sql/src/pure.rs

```diff
@@ -207,20 +207,31 @@ pub async fn purify_create_source(
                     _ => bail!("{} is not a postgres connection", item.name()),
                 }
             };
+            let crate::plan::statement::PgConfigOptionExtracted { publication, .. } =
+                options.clone().try_into()?;
+            let publication = publication.expect("verified existed");
```

Comment:
Is this true? Is this going to cause the same bug that we just fixed with Kafka connections not specifying their `TOPIC`?